builder/amazon/common/step_ami_region_copy.go

```diff
@@ -64,7 +64,7 @@ func (s *StepAMIRegionCopy) Run(ctx context.Context, state multistep.StateBag) m
 	ui := state.Get("ui").(packer.Ui)
 	amis := state.Get("amis").(map[string]string)
 	snapshots := state.Get("snapshots").(map[string][]string)
-	intermediary := state.Get("intermediary_image").(bool)
+	intermediary, _ := state.Get("intermediary_image").(bool)
```

Comment:
That adds some safety by assigning the default nil value for the given type if anything goes wrong with the *type assertion* e.g. in the event `intermediary_image` is nil, instead of a panic, we end up with intermediary being assigned a value of `false`.

It's easy to mix up what's happening here. As 'Packer people' we are *very* used to seeing the same pattern with `state.GetOK` e.g.

```
if foo, ok := state.GetOK("somekey"); ok {
    // do something with foo
fi
```

Clearly, the `ok` here is assigned the second value returned by the `GetOk` method.

However, trying to use a type assertion with `GetOk`:

```
mybool, _ := state.GetOk("somekey").(bool)
```

... fails with "multiple-value state.GetOk() in single-value context". The error tells us Go doesn't know what to do with the second value returned by `GetOk` as the second variable here (in this instance `_`) is being assigned the return value of the *type assertion*.

Reply:
> I'm surprised this is enough.

You're not alone there! It doesn't look like a lot does it!! 😄 

However, I've tested a build with this change in place and it fixed the issue for me!

Reply:
Yep, on second glance I realized the extra value was coming from the type assertion, not the Get. Just took me a minute. This is very nice and clean. I added a test to make sure we don't regress on this.